test/unit/reporters/html/view-model.js

```diff
@@ -43,10 +43,23 @@ describe('ViewModel', () => {
 
         model.addSuccess(stubTest_({
             suite: {fullUrl: '/test/url'}
-        }));
+        }), {failedOnly: false});
 
         const metaInfo = JSON.parse(getModelResult_(model).metaInfo);
 
         assert.equal(metaInfo.url, '/test/url');
     });
+
+    it('should not prepare "metaInfo", when "failedOnly" is true', () => {
+        const model = mkViewModel_();
+
+        model.addSuccess(stubTest_({
+            suite: {
+                file: '/path/file.js',
+                fullUrl: '/test/url'
+            }
+        }), {failedOnly: true});
+
+        assert.isUndefined(model.getResult().suites);
+    });
```

Comment:
we should add one more test here to check that counter is increased even if `failedOnly` is `true`.
For this you should stub method `onPassed` for `TestCounter` module